frame/society/src/lib.rs

```diff
@@ -0,0 +1,792 @@
+		let candidates = Self::take_selected(pot);
+		<Candidates<T>>::put(&candidates);
+
+		// Select sqrt(n) random members from the society and make them skeptics.
+		let pick_member = |_| pick_item(&mut rng, &members[..]).expect("exited if members empty; qed");
+		for skeptic in (0..members.len().integer_sqrt()).map(pick_member) {
+			for (_, c, _) in candidates.iter() {
+				<Votes<T>>::insert(c, skeptic, Vote::Skeptic);
+			}
+		}
+	}
+
+	/// Attempt to slash the payout of some member. Return the payout block number most in the
+	/// future along with the total amount deducted.
+	fn slash_payout(who: &T::AccountId, value: BalanceOf<T>) -> (T::BlockNumber, BalanceOf<T>) {
+		let mut latest = T::BlockNumber::zero();
+		let mut rest = value;
+		let mut payouts = <Payouts<T>>::get(who);
+		if !payouts.is_empty() {
+			let mut dropped = 0;
+			for (when, amount) in payouts.iter_mut() {
+				latest = *when;
+				if let Some(new_rest) = rest.checked_sub(&amount) {
+					// not yet totally slashed after this one; drop it completely.
+					rest = new_rest;
+					dropped += 1;
+				} else {
+					// whole slash is accounted for.
+					*amount -= rest;
+					rest = Zero::zero();
+					break;
+				}
+			}
+			<Payouts<T>>::insert(who, &payouts[dropped..]);
+		}
+		(latest, value - rest)
+	}
+
+	/// Bump the payout amount of `who`, to be unlocked at the given block number.
+	fn bump_payout(who: &T::AccountId, when: T::BlockNumber, value: BalanceOf<T>) {
+		<Payouts<T>>::mutate(who, |payouts| match payouts.binary_search_by_key(&when, |x| x.0) {
+			Ok(index) => payouts[index].1 += value,
+			Err(index) => payouts.insert(index, (when, value)),
+		});
+	}
+
+	/// Suspend a user, removing them from the member list.
+	fn suspend_member(who: &T::AccountId) {
+		if Self::remove_member(&who).is_ok() {
+			<SuspendedMembers<T>>::insert(who, ());
+			<Strikes<T>>::remove(who);
+		}
+	}
+
+	/// Pay an accepted candidate their bid value.
+	fn pay_accepted_candidate(
+		candidate: &T::AccountId,
+		value: BalanceOf<T>,
+		kind: BidKind<T::AccountId, BalanceOf<T>>,
+		maturity: T::BlockNumber,
+	) {
+		let value = match kind {
+			BidKind::Deposit(deposit) => {
+				// In the case that a normal deposit bid is accepted we unreserve
+				// the deposit.
+				let _ = T::Currency::unreserve(candidate, deposit);
+				value
+			}
+			BidKind::Vouch(who, tip) => {
+				// In the case that a vouched-for bid is accepted we unset the
+				// vouching status and transfer the tip over to the voucher.
+				Self::bump_payout(&who, maturity, tip);
+				// Really shouldn't fail given the conditional we're in.
+				let _ = Self::unset_vouching(&who);
+				value.saturating_sub(tip)
+			}
+		};
+
+		Self::bump_payout(candidate, maturity, value);
+	}
+
+	/// Remove a suspended candidate. Slash their deposit or give strike to their voucher.
+	fn slash_suspended_candidate(who: &T::AccountId) {
+		if let Some((_, kind)) = <SuspendedCandidates<T>>::take(who) {
```

Comment:
this is an unnecessary pessimisation. just pass in the value at call and use `::remove` instead.

Reply:
I got rid of this function and just moved the logic into the code. Thanks for the sharp eyes.